packages/cli/src/commands/set-admin.ts

```diff
@@ -19,46 +43,111 @@ const register: (program: any) => any = (program) => program
   .description(description)
   .option('-f, --force', 'bypass a manual check')
   .withNetworkOptions()
+  .withNonInteractiveOption()
   .action(action);
 
-async function action(contractFullNameOrAddress: string, newAdmin: string, options: any): Promise<void | never> {
-  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration(options);
+async function action(contractReference: string, newAdmin: string, options: any): Promise<void | never> {
+  const { force, network: networkInOpts, interactive  } = options;
+  const networkOpts = await promptForNetwork(options, () => getCommandProps());
+
+  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration({ ...options, ...networkOpts });
   if (!await hasToMigrateProject(network)) process.exit(0);
 
-  let proxyAddress;
-  let contractAlias;
-  let packageName;
+  const { proxyReference, newAdmin: pickedNewAdmin } = await promptForProxies(contractReference, newAdmin, network, options);
+  const parsedContractReference = parseContractReference(proxyReference);
 
-  if (!contractFullNameOrAddress) throw Error('You have to specify at least a new admin address.');
+  if (!parsedContractReference) throw Error('You have to specify at least a new admin address.');
 
-  // we assume if newAdmin is empty it was specified as first argument
-  if (!newAdmin) {
-    newAdmin = contractFullNameOrAddress;
-    contractFullNameOrAddress = '';
-  }
-
-  if (!options.force) {
+  // has to be a standalone question from interactivity
+  // because it is security related and can't be disabled with interactivity set to false
+  if (!force) {
     const answers = await inquirer.prompt([
```

Comment:
Same as above. Let's try only to prompt the user when they don't specify the `--no-interactive` flag. That being saind, I believe we can implement this question as all the other ones (i.e., setting the question's properties in the `setCommandProps()` function).

Reply:
It is not related to `interactivity` as a comment says. It is a security feature thus it can't be disabled with `interactivity` flag. Also there is no `getCommandProps` function I assume you mean `getCommandProps` function.

Reply:
My point is that we can still have the security feature without prompting a question to the user and just fail if no `--force` flag is provided.

Reply:
I like idea failing with  `--no-interactive` && `force` not set.